test/vstest.console.UnitTests/Processors/CLIRunSettingsArgumentProcessorTests.cs

```diff
@@ -347,5 +347,46 @@ public static IEnumerable<object[]> TestRunParameterArgValidTestCases()
             },
         };
         #endregion
+
+        [TestMethod]
+        public void InitializeShouldMergeTestRunParametersWithSpaces()
+        {
+            // in powershell call: ConsoleApp1.exe --% --TestRunParameters.Parameter(name =\"myParam\", value=\"myValue\")
+            // args:
+            //--
+            //TestRunParameters.Parameter(name = "myParam",
+            //value = "myValue")
+
+            // in cmd: ConsoleApp1.exe -- TestRunParameters.Parameter(name=\"myParam\", value=\"myValue\")
+            // args:
+            //--
+            //TestRunParameters.Parameter(name = "myParam",
+            //value = "myValue")
+
+            // in ubuntu wsl without escaping the space: ConsoleApp1.exe-- TestRunParameters.Parameter\(name =\"myParam\", value=\"myValue\"\)
+            // args:
+            //--
+            //TestRunParameters.Parameter(name = "myParam",
+            //value = "myValue")
+
+            // in ubuntu wsl with escaping the space: ConsoleApp1.exe-- TestRunParameters.Parameter\(name =\"myParam\",\ value=\"myValue\"\)
+            // args:
+            //--
+            //TestRunParameters.Parameter(name = "myParam", value = "myValue")
+
+            var args = new string[] {
+                "--",
+                "TestRunParameters.Parameter(name=\"myParam\",",
+                "value=\"myValue\")",
+                "TestRunParameters.Parameter(name=\"myParam2\",",
+                "value=\"myValue 2\")",
+            };
+
+            var runsettings = "<?xml version=\"1.0\" encoding=\"utf-16\"?>\r\n<RunSettings>\r\n  <DataCollectionRunSettings>\r\n    <DataCollectors />\r\n  </DataCollectionRunSettings>\r\n  <TestRunParameters>\r\n    <Parameter name=\"myParam\" value=\"myValue\" />\r\n    <Parameter name=\"myParam2\" value=\"myValue 2\" />\r\n  </TestRunParameters>\r\n</RunSettings>";
```

Comment:
I doubt stringbuilder vs string concatenation matters in this case. But yeah formatting it would be nice. the easiest way is imho to make it an array of strings and then string.join(environment.newline). It's also a nicer syntax.

Reply:
Used string join over an array: 

![image](https://user-images.githubusercontent.com/5735905/79974891-5b4fae00-849a-11ea-9e22-979e37d97bab.png)